vibi-dpu/src/core/relevance.rs

```diff
@@ -158,3 +157,39 @@ fn comment_text(relevance_vec: &Vec<Relevance>, auto_assign: bool) -> String {
 
     return comment;
 }
+
+fn deduplicated_relevance_vec_for_comment(relevance_vec: &Vec<Relevance>) -> (HashMap<Vec<String>, f32>, Vec<String>) {
+    let mut combined_relevance_map: HashMap<Vec<String>, f32> = HashMap::new();
+    let mut unmapped_aliases = Vec::new();
+
+    // Iterate through relevance_vec and handle entries with provider IDs
+    for relevance_obj in relevance_vec {
+        let provider_ids_opt = relevance_obj.handles();
+        if let Some(provider_ids) = provider_ids_opt {
+            // Check if combined relevance for handles set already exists
+            let mut found = false;
+            for (existing_handles, relevance) in combined_relevance_map.iter_mut() {
+                let intersection: HashSet<_> = existing_handles.iter().cloned().collect();
+                if !intersection.is_empty() && provider_ids.iter().any(|h| intersection.contains(h)) {
+                    *relevance += relevance_obj.relevance_num(); // Add relevance to existing combined relevance
+                    found = true;
+                    break;
+                }
+            }
+
+            // If no combined relevance found, add a new entry
+            if !found {
+                combined_relevance_map.insert(provider_ids.clone(), relevance_obj.relevance_num());
+            }
+        } else {
+            // For entries without provider IDs, add them to the combined_relevance map
+            let git_alias = relevance_obj.git_alias();
+            let git_alias_vec: Vec<String> = vec![git_alias.to_owned()];
+            combined_relevance_map.insert(git_alias_vec, relevance_obj.relevance_num());
+            // Add the git alias to the unmapped aliases array
+            unmapped_aliases.push(git_alias.to_string());
+        }
+    }
+
+    (combined_relevance_map, unmapped_aliases)
```

Comment:
@MuskanPaliwal - Point 1 and 2 seem relevant to me

Reply:
@coderabbit - can you give us an alternate approach that addresses all the concerns you raised in this conversation

Reply:
creating an issue and merging this pr for now, since this change is not a priority. Why? - handles array will have at most 4-5 entries, probably less than 10 at all times.